<details>
  <summary><font size="3" color="gold"><b>  HDDA Assignment 1 </b></font></summary>

<details>
  <summary><font size="3" color="gold"><b>James Wandiya </b></font></summary>

<span style="color: gold">Answer 2</span>

<a name="1"> # Packages </a>



In [2]:
import numpy as np
from scipy import stats
import pandas as pd

<span style="color: gold">Data</span>

In [15]:
Data = pd.read_excel('lab1.xlsx', sheet_name='Ex1') 

In [16]:
Data

How many people do you follow on Instagram?  \
0                                            126   
1                                            700   
2                                            275   
3                                            500   
4                                            188   
..                                           ...   
364                                          360   
365                                          344   
366                                          218   
367                                          312   
368                                            0   

     How many followers do you have on Instagram ?  
0                                              161  
1                                              380  
2                                              207  
3                                              900  
4                                              160  
..                                             ...  
364                                            392  
365                                            370  
366                                            217  
367                                            280  
368                                              0  

[369 rows x 2 columns]

In [17]:
x_values = Data.iloc[:, 0]
y_values = Data.iloc[:, 1]

In [18]:
p_x = np.array(x_values, dtype=float)
f_y = np.array(y_values, dtype=float)

<details>
  <summary><font size="3" color="gold"><b>To Calculate correlation coefficient</b></font></summary>

In [19]:
r = np.corrcoef(p_x, f_y)[0, 1]
print(f"Correlation coefficient: r = {r:.4f}")

Correlation coefficient: r = 0.6412


<details>
  <summary><font size="3" color="gold"><b>To Calculate regression equation</b></font></summary>

In [20]:
slope, intercept, r_value, p_value, std_err = stats.linregress(p_x, f_y)

print(f"   ŷ = {intercept:.2f} + {slope:.4f}x")
print(f"   or to write like this: Followers = {intercept:.2f} + {slope:.4f} × (People Followed)")

   ŷ = 60.35 + 0.8937x
   or to write like this: Followers = 60.35 + 0.8937 × (People Followed)


<details>
  <summary><font size="3" color="gold"><b>To Forecast Instagram followers of an AIMS student who follows 200
people.</b></font></summary>

In [21]:

y_pred = intercept + slope * 200
print(f"Prediction for 200: {y_pred:.4f}")

Prediction for 200: 239.0820


<span style="color: gold">Insight derived:</span> The model shows that while following more people is associated with having more followers, the relationship is only moderately strong.

<span style="color: gold">Answer 3</span>

<span style="color: gold">Data</span>

In [3]:
Data2 = pd.read_excel('lab1.xlsx', sheet_name='Ex2')

In [4]:
Data2

NbOfContacts  Sales
0            14     24
1            12     14
2            20     28
3            16     30
4            46     80
5            23     30
6            48     90
7            50     85
8            55    120
9            50    110

In [8]:
x_values2 = Data2.iloc[:, 0]
y_values2 = Data2.iloc[:, 1]

In [10]:
#Independent variable (# NbOfContacts)
#Dependent variable (# Sales)
x = np.array(x_values2, dtype=float)
y = np.array(y_values2, dtype=float)

<details>
  <summary><font size="3" color="gold"><b> (a) Determine the regression equation. </b></font></summary>

In [11]:
slope, intercept, r, p, std_err = stats.linregress(x, y)

In [22]:
print(f"   Estimated model: ŷ = {intercept:.3f} + {slope:.3f}x")
print(f"   or: Sales = {intercept:.3f} + {slope:.3f} × Contacts")

print(f"   The estimated intercept: β₀ = {intercept:.3f}")
print(f"   The estimated slope: β₁ = {slope:.3f}")


   Estimated model: ŷ = 60.350 + 0.894x
   or: Sales = 60.350 + 0.894 × Contacts
   The estimated intercept: β₀ = 60.350
   The estimated slope: β₁ = 0.894


<details>
  <summary><font size="3" color="gold"><b> (b)Estimated sales for 40 contacts </b></font></summary>

In [13]:
contacts = 40
predicted_sales = intercept + slope * contacts


In [24]:
print(f" For {contacts} contacts: The predicted sales = {predicted_sales:.3f}")

 For 40 contacts: The predicted sales = 75.585


<span style = "color: gold" >This means that: </span> If 40 contacts are made, estimated sales are 75.6(Thousand dollar)